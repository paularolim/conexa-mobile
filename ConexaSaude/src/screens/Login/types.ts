import { Control } from 'react-hook-form';
import * as yup from 'yup';

import { schema } from './validation';

export type LoginSchema = yup.InferType<typeof schema>;

export interface LoginViewProps {
  loading: boolean;
  errorMessage: string | null;
  handleDismissKeyboard: () => void;
  control: Control<LoginSchema, any>;
  onSubmit: (e?: React.BaseSyntheticEvent<object, any, any> | undefined) => Promise<void>;
}
