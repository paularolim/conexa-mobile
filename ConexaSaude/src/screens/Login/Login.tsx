import { View } from 'react-native';

import { Button } from '@components/Button';
import { Text } from '@components/Text';
import { useAuth } from '@hooks/useAuth';

export function Login() {
  const { handleLogin } = useAuth();

  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Text>Login</Text>
      <Button onPress={handleLogin}>Entrar</Button>
    </View>
  );
}
