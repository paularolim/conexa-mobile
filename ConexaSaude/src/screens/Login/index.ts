export * from './Login';
export * from './LoginView';
export * from './types';
export * from './useLoginViewModel';
export * from './validation';
