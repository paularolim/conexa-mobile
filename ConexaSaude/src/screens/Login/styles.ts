import { Image, View } from 'react-native';
import styled from 'styled-components/native';

import { Text } from '@components/Text';

export const Container = styled(View)`
  background-color: white;
  flex: 1;
  justify-content: center;
  align-items: center;
  padding: 0px 24px;
  gap: 20px;
`;

export const Section = styled(View)<{ space?: number }>`
  width: 100%;
  gap: ${({ space }) => space || 0}px;
`;

export const Logo = styled(Image)`
  max-width: 60%;
  resize-mode: contain;
`;

export const Title = styled(Text)`
  font-size: 20px;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.gray};
`;
