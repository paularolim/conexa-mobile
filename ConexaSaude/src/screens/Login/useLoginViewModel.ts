/* eslint-disable no-console */
import { useState } from 'react';
import { Keyboard } from 'react-native';

import { useAuth } from '@hooks/useAuth';
import { loginRepository } from '@repositories/login-repository';

export function useLoginViewModel() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const { handleSaveData } = useAuth();

  const handleLogin = () => {
    if (!email && !password) {
      // TODO: implement validation
      console.log('handleLogin - email and password is required');
      return;
    }

    loginRepository({ email, password })
      .then((data) => {
        if (data?.data.token && data?.data.nome && data?.data.email) {
          handleSaveData({
            token: data?.data.token,
            email: data?.data.email,
            name: data?.data.nome,
          });
        } else {
          // TODO: error feedback
          console.log('handleLogin - error on login');
        }
      })
      .catch((reason) => {
        // TODO: error feedback
        console.log(`handleLogin - ${reason}`);
      });
  };

  const handleDismissKeyboard = () => {
    Keyboard.dismiss();
  };

  return { handleLogin, handleDismissKeyboard, email, setEmail, password, setPassword };
}
