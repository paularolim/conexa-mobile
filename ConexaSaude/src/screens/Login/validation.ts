import * as yup from 'yup';

export const schema = yup
  .object({
    email: yup.string().email('Digite um e-mail válido').required('E-mail não informado!'),
    password: yup.string().required('Senha não informada!'),
  })
  .required();
