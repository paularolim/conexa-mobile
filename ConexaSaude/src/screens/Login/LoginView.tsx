import { TouchableWithoutFeedback } from 'react-native';

import logo from '@assets/images/logo.png';
import { Button } from '@components/Button/Button';
import { Divider } from '@components/Divider/Divider';
import { Input } from '@components/Input';

import { Container, Logo, Section, Title } from './styles';
import { LoginViewProps } from './types';

export function LoginView({
  email,
  password,
  setEmail,
  setPassword,
  handleLogin,
  handleDismissKeyboard,
}: LoginViewProps) {
  return (
    <TouchableWithoutFeedback onPress={handleDismissKeyboard}>
      <Container>
        <Section>
          <Logo source={logo} />
        </Section>

        <Section space={8}>
          <Title>Para entrar, utilize as credenciais que vocÃª recebeu via e-mail</Title>
          <Divider />
        </Section>

        <Section space={16}>
          <Input
            label="E-mail"
            keyboardType="email-address"
            autoCapitalize="none"
            value={email}
            onChangeText={setEmail}
          />
          <Input
            label="Senha"
            secureTextEntry
            hasPasswordIcon
            autoCapitalize="none"
            value={password}
            onChangeText={setPassword}
          />

          <Button onPress={handleLogin}>Entrar</Button>
          <Button type="ghost">Esqueci a senha</Button>
        </Section>
      </Container>
    </TouchableWithoutFeedback>
  );
}
