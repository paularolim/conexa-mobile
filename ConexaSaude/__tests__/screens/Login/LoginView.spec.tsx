import renderer from 'react-test-renderer';

import { LoginView } from '@screens/Login/LoginView';

import { Wrapper } from '../../Wrapper';

describe('LoginView', () => {
  it('should render screen correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <LoginView handleLogin={jest.fn()} handleDismissKeyboard={jest.fn()} />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
