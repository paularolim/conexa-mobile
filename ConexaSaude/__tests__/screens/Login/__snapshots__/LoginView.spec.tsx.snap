// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView should render screen correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "gap": 20,
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "gap": 0,
          "width": "100%",
        },
      ]
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/logo.png",
        }
      }
      style={
        [
          {
            "maxWidth": "60%",
            "resizeMode": "contain",
          },
        ]
      }
    />
  </View>
  <View
    space={8}
    style={
      [
        {
          "gap": 8,
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
          },
          {
            "color": "#707683",
            "fontSize": 20,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Para entrar, utilize as credenciais que vocÃª recebeu via e-mail
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
            "gap": 10,
          },
        ]
      }
    >
      <View
        primary={true}
        style={
          [
            {
              "backgroundColor": "#F28080",
              "borderRadius": 2,
              "height": 5,
              "width": 80,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#C2CFE0",
              "borderRadius": 2,
              "height": 5,
              "width": 80,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    space={16}
    style={
      [
        {
          "gap": 16,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "backgroundColor": "#FFFFFF",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "color": "#707683",
              "fontSize": 18,
            },
          ]
        }
      >
        E-mail
      </Text>
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#707683",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "minHeight": 48,
              "paddingBottom": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 8,
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          keyboardType="email-address"
          style={
            [
              {
                "color": "#000000",
                "flexGrow": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          testID="field"
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "backgroundColor": "#FFFFFF",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "color": "#707683",
              "fontSize": 18,
            },
          ]
        }
      >
        Senha
      </Text>
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#707683",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "minHeight": 48,
              "paddingBottom": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 8,
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          secureTextEntry={true}
          style={
            [
              {
                "color": "#000000",
                "flexGrow": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          testID="field"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="password-visibility-toggle"
        >
          <svg
            data-file-name="SvgEyeOn"
            height={25}
            stroke="#707683"
            width={25}
          />
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2ED47A",
          "borderRadius": 4,
          "justifyContent": "center",
          "maxWidth": "100%",
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "color": "#FFFFFF",
              "fontSize": 18,
            },
          ]
        }
        type="filled"
      >
        Entrar
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 4,
          "justifyContent": "center",
          "maxWidth": "100%",
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
            },
            {
              "color": "#109CF1",
              "fontSize": 18,
            },
          ]
        }
        type="ghost"
      >
        Esqueci a senha
      </Text>
    </View>
  </View>
</View>
`;
